riot/riot.py

```diff
@@ -309,6 +312,55 @@ def is_warning(self, output):
         lower_output = output.lower()
         return any(warning in lower_output for warning in self.warnings)
 
+    def iter_selected_venvs(
+        self,
+        pattern: t.Pattern[str],
+        venv_pattern: t.Pattern[str],
+        pythons: t.Optional[t.Set[Interpreter]] = None,
+        skip_missing: bool = False,
+    ) -> t.Generator["VenvInstance", None, None]:
+        def _iter_venvs() -> t.Generator["VenvInstance", None, None]:
+            for inst in self.venv.instances(pattern=pattern):
+                if pythons and inst.py not in pythons:
+                    logger.debug(
+                        "Skipping venv instance %s due to interpreter mismatch", inst
+                    )
+                    continue
+
+                try:
+                    venv_path = inst.venv_path()
+                except FileNotFoundError:
+                    if skip_missing:
+                        logger.warning("Skipping missing interpreter %s", inst.py)
+                        continue
+                    else:
+                        raise
+
+                if not venv_pattern.search(venv_path):
+                    logger.debug(
+                        "Skipping venv instance '%s' due to pattern mismatch", venv_path
+                    )
+                    continue
+
+                yield inst
+
+        ci_node_total = os.environ.get("CI_NODE_TOTAL")
+        ci_node_index = os.environ.get("CI_NODE_INDEX")
+        if os.environ.get("CIRCLECI") == "true":
+            if "CIRCLE_NODE_TOTAL" in os.environ and "CIRCLE_NODE_INDEX" in os.environ:
+                ci_node_total = os.environ["CIRCLE_NODE_TOTAL"]
+                ci_node_index = os.environ["CIRCLE_NODE_INDEX"]
+
+        if ci_node_total and ci_node_index:
+            venvs = list(_iter_venvs())
```

Comment:
I was originally thinking we might want to do `venvs = sorted(_iter_venvs())` to ensure the results are consistent, but would it be weird that the ordering of the venvs changes based on whether you use parallelism or not?